Engine/Annotations/Annotation.cs

```diff
@@ -1810,18 +1791,23 @@ public void Write(object source)
                         rdonly = true;
                     if (!rdonly)
                     {
-                        lst2.Clear();
-
+                        // Arrays must be re-allocated
```

Comment:
Should these changes be in a separate PR?

Reply:
Hmm, yes. I wanted to sneak in a nice simplification but I'll open a separate PR